### ISSUE TITLE
Ajax Convertion not supporting locales

### ISSUE THREAD

**Issue by sergiocastelani** [kartik-v/yii2-datecontrol#54]
Seems that [DateTime::createFromFormat](http://php.net/manual/en/datetime.createfromformat.php), used at [ParseController::actionConvert()](https://github.com/kartik-v/yii2-datecontrol/blob/master/controllers/ParseController.php) doesn't take LC_TIME in consideration.

So when displaying a widget with a displayFormat like **php:d / F / Y** (fulltext month) and a saveFormat like **php:d-m-Y**, the conversion will fail because PHP cant find the month name for languages other then 'en'.

I could fix it by replacing the code using **DateTime::createFromFormat**:

```php
if ($dispTimezone != null) {
    $date = DateTime::createFromFormat($dispFormat, $dispDate, new DateTimeZone($dispTimezone));
} else {
    $date = DateTime::createFromFormat($dispFormat, $dispDate);
}
```
with:

```php
$fmt = new \IntlDateFormatter(Yii::$app->formatter->locale, \IntlDateFormatter::FULL, \IntlDateFormatter::FULL);
$fmt->setPattern(FormatConverter::convertDatePhpToIcu($dispFormat));
$fmt->setTimezone($dispTimezone);
$date = new DateTime();
$date->setTimestamp($fmt->parse($dispDate));
```

**Comment by kartik-v** [comment]
The `IntlDateFormatter` requires php `intl` extension to be installed - else it will not work. So this is not a global solution for developers who do not have this installed on their webservers (and who do not need such locales).

Will check later on a generic approach for this to address both.

**Comment by kartik-v** [comment]
Resolved via #66 without using `IntlDateFormatter` ... read why.
